fusequery/query/src/servers/clickhouse/clickhouse_stream.rs

```diff
@@ -12,24 +12,31 @@ use common_arrow::arrow::array::*;
 use common_arrow::arrow::datatypes::*;
 use common_datablocks::DataBlock;
 use common_datavalues::DataArrayRef;
+use common_datavalues::DataSchemaRef;
 use common_exception::ErrorCodes;
 use common_exception::Result;
 use common_streams::SendableDataBlockStream;
 use futures::stream::Stream;
 use futures::StreamExt;
 
 pub struct ClickHouseStream {
-    input: SendableDataBlockStream
+    input: SendableDataBlockStream,
+    block_index: usize,
+    schema: DataSchemaRef
 }
 
 impl ClickHouseStream {
-    pub fn create(input: SendableDataBlockStream) -> Self {
-        ClickHouseStream { input }
+    pub fn create(input: SendableDataBlockStream, schema: DataSchemaRef) -> Self {
+        ClickHouseStream {
+            input,
+            block_index: 0,
+            schema
+        }
     }
 
     pub fn convert_block(&self, block: DataBlock) -> Result<ClickHouseBlock> {
         let mut result = ClickHouseBlock::new();
-        if block.is_empty() {
+        if block.is_empty() && block.num_columns() == 0 {
```

Comment:
is_empty() includes rows and columns zero check:
https://github.com/datafuselabs/datafuse/blob/0136e6452dc0245ae33f7264a0852f277f13bc53/common/datablocks/src/data_block.rs#L54-L72